logrotate/src/LogRotateHelper.ipp

```diff
@@ -403,9 +427,9 @@ bool LogRotateHelper::createCompressedFile(std::string file_name, std::string gz
         return false;
     }
 
-    int N;
+    size_t N;
     while ((N = fread(buffer, 1, buffer_size, input)) > 0) {
-        gzwrite(output, buffer, N);
+        gzwrite(output, buffer, static_cast<unsigned int>(N));
```

Comment:
Well,

In windows it depends. So if you are compiling for x64 it is not...

#ifdef _WIN64
typedef __int64             ptrdiff_t;
typedef unsigned __int64    size_t;
#else
typedef _W64 int            ptrdiff_t;
typedef _W64 unsigned int   size_t;
#endif

On 15 December 2015 at 15:13, Kjell Hedström <notifications@github.com>
wrote:

> In logrotate/src/LogRotateHelper.ipp
> <https://github.com/KjellKod/g3sinks/pull/20#discussion_r47691314>:
>
> >      while ((N = fread(buffer, 1, buffer_size, input)) > 0) {
> > -        gzwrite(output, buffer, N);
> > +        gzwrite(output, buffer, static_cast<unsigned int>(N));
>
> isn't size_t equivalent to unsigned int on your system?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KjellKod/g3sinks/pull/20/files#r47691314>.
>